pkg/cli/startbuild/startbuild.go

```diff
@@ -513,6 +526,120 @@ func (o *StartBuildOptions) streamBuildLogs(build *buildv1.Build) error {
 	return err
 }
 
+// checkNonExistantResources checks for Resources (config maps, secrets, etc) listed in the BuildConfig
+// and returns error in case any of them are missing, or in case of unexpected behavior
+func (o *StartBuildOptions) checkNonExistantResources() error {
+	buildConfig, err := o.BuildClient.BuildConfigs(o.Namespace).Get(context.TODO(), o.Name, metav1.GetOptions{})
```

Comment:
This isn't a blocking issue, more an observation that we have a lot of `context.TODO()` calls. Ideally we should have a parent context for the start-build command. We can then create sub-contexts which then give us flexibility to add things like cancellation and timeouts.

Reply:
I'll do it for the `startbuild.go` then. Ideally we should create an issue and work on that through as much as possible, at least whole `build` part of `oc`. I don't want to do more here in this PR at least, in order to not make it bigger that it has to be.